web/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,69 @@
+import {LeftOutlined, RightOutlined} from '@ant-design/icons';
+import {Button} from 'antd';
+import {ReactNode} from 'react';
+
+import * as S from './Pagination.styled';
+
+interface IProps {
+  children: ReactNode;
+  emptyComponent?: ReactNode;
+  hasNext: boolean;
+  hasPrev: boolean;
+  isEmpty: boolean;
+  isFetching: boolean;
+  isLoading: boolean;
+  loadingComponent?: ReactNode;
+  loadNext(): void;
+  loadPrev(): void;
+}
+
+const Pagination = ({
+  children,
+  emptyComponent,
+  hasNext,
+  hasPrev,
+  isEmpty,
+  isFetching,
+  isLoading,
+  loadingComponent,
+  loadNext,
+  loadPrev,
+}: IProps) => {
+  const handleNextPage = () => {
+    if (isLoading || !hasNext) return;
```

Comment:
Instead of having this logic here, why we don't just disable the buttons based on the same condition?